ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/LimitConnectionsJsonRpcHttpServiceTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.api.jsonrpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import org.junit.Test;
+
+public class LimitConnectionsJsonRpcHttpServiceTest extends JsonRpcHttpServiceTestBase {
+
+  @Test
+  public void limitActiveConnections() throws Exception {
+    OkHttpClient newClient = new OkHttpClient();
+    int i;
+    for (i = 0; i < JsonRpcConfiguration.DEFAULT_MAX_ACTIVE_CONNECTIONS; i++) {
```

Comment:
Is it possible to run this test with a lower limit? Probably 1? Then the second connection should fail.

It seems to be a waste of resources (and test time) opening 80 real connections to test the limit if we can reduce the config before starting the server.

Reply:
refactored to add this